### ISSUE TITLE
SQL Statement Query error after SLOW API CALL DETECTED: notes/user-list-timeline

### ISSUE THREAD

**Issue by ghost** [syuilo/Misskey#9116]
## 💡 Summary

I am receiving regular messages with 

> misskey-web-1    | ERR  1	[api]	Internal error occurred in notes/user-list-timeline: canceling statement due to statement timeout
misskey-web-1    | WARN 1	[api]	SLOW API CALL DETECTED: notes/user-list-timeline (10033.614974975586ms)

when using the timeline widget and want to display a user list.  reloading fixes the problem. I am unsure what the problem is. Disk IO seems fine. There is no wait or cpu issue.

## 🥰 Expected Behavior

No problem when using lists with the timeline widget

## 🤬 Actual Behavior

The docker log for the web container shows the sql statement

> misskey-db-1     | 2022-10-08 09:49:09.739 UTC [22563] STATEMENT:  SELECT DISTINCT "distinctAlias"."note_id" AS "ids_note_id", "distinctAlias"."note_id" FROM (SELECT "note"."id" AS "note_id", "note"."createdAt" AS "note_createdAt", "note"."replyId" AS "note_replyId", "note"."renoteId" AS "note_renoteId", "note"."threadId" AS "note_threadId", "note"."text" AS "note_text", "note"."name" AS "note_name", "note"."cw" AS "note_cw", "note"."userId" AS "note_userId", "note"."localOnly" AS "note_localOnly", "note"."renoteCount" AS "note_renoteCount", "note"."repliesCount" AS "note_repliesCount", "note"."reactions" AS "note_reactions", "note"."visibility" AS "note_visibility", "note"."uri" AS "note_uri", "note"."url" AS "note_url", "note"."fileIds" AS "note_fileIds", "note"."attachedFileTypes" AS "note_attachedFileTypes", "note"."visibleUserIds" AS "note_visibleUserIds", "note"."mentions" AS "note_mentions", "note"."mentionedRemoteUsers" AS "note_mentionedRemoteUsers", "note"."emojis" AS "note_emojis", "note"."tags" AS "note_tags", "note"."hasPoll" AS "note_hasPoll", "note"."channelId" AS "note_channelId", "note"."userHost" AS "note_userHost", "note"."replyUserId" AS "note_replyUserId", "note"."replyUserHost" AS "note_replyUserHost", "note"."renoteUserId" AS "note_renoteUserId", "note"."renoteUserHost" AS "note_renoteUserHost", "user"."id" AS "user_id", "user"."createdAt" AS "user_createdAt", "user"."updatedAt" AS "user_updatedAt", "user"."lastFetchedAt" AS "user_lastFetchedAt", "user"."lastActiveDate" AS "user_lastActiveDate", "user"."hideOnlineStatus" AS "user_hideOnlineStatus", "user"."username" AS "user_username", "user"."name" AS "user_name", "user"."followersCount" AS "user_followersCount", "user"."followingCount" AS "user_followingCount", "user"."notesCount" AS "user_notesCount", "user"."avatarId" AS "user_avatarId", "user"."bannerId" AS "user_bannerId", "user"."tags" AS "user_tags", "user"."isSuspended" AS "user_isSuspended", "user"."isSilenced" AS "user_isSilenced", "user"."isLocked" AS "user_isLocked", "user"."isBot" AS "user_isBot", "user"."isCat" AS "user_isCat", "user"."isAdmin" AS "user_isAdmin", "user"."isModerator" AS "user_isModerator", "user"."isExplorable" AS "user_isExplorable", "user"."isDeleted" AS "user_isDeleted", "user"."emojis" AS "user_emojis", "user"."host" AS "user_host", "user"."inbox" AS "user_inbox", "user"."sharedInbox" AS "user_sharedInbox", "user"."featured" AS "user_featured", "user"."uri" AS "user_uri", "user"."followersUri" AS "user_followersUri", "user"."showTimelineReplies" AS "user_showTimelineReplies", "user"."token" AS "user_token", "user"."driveCapacityOverrideMb" AS "user_driveCapacityOverrideMb", "avatar"."id" AS "avatar_id", "avatar"."createdAt" AS "avatar_createdAt", "avatar"."userId" AS "avatar_userId", "avatar"."userHost" AS "avatar_userHost", "avatar"."md5" AS "avatar_md5", "avatar"."name" AS "avatar_name", "avatar"."type" AS "avatar_type", "avatar"."size" AS "avatar_size", "avatar"."comment" AS "avatar_comment", "avatar"."blurhash" AS "avatar_blurhash", "avatar"."properties" AS "avatar_properties", "avatar"."storedInternal" AS "avatar_storedInternal", "avatar"."url" AS "avatar_url", "avatar"."thumbnailUrl" AS "avatar_thumbnailUrl", "avatar"."webpublicUrl" AS "avatar_webpublicUrl", "avatar"."webpublicType" AS "avatar_webpublicType", "avatar"."accessKey" AS "avatar_accessKey", "avatar"."thumbnailAccessKey" AS "avatar_thumbnailAccessKey", "avatar"."webpublicAccessKey" AS "avatar_webpublicAccessKey", "avatar"."uri" AS "avatar_uri", "avatar"."src" AS "avatar_src", "avatar"."folderId" AS "avatar_folderId", "avatar"."isSensitive" AS "avatar_isSensitive", "avatar"."maybeSensitive" AS "avatar_maybeSensitive", "avatar"."maybePorn" AS "avatar_maybePorn", "avatar"."isLink" AS "avatar_isLink", "avatar"."requestHeaders" AS "avatar_requestHeaders", "avatar"."requestIp" AS "avatar_requestIp", "banner"."id" AS "banner_id", "banner"."createdAt" AS "banner_createdAt", "banner"."userId" AS "banner_userId", "banner"."userHost" AS "banner_userHost", "banner"."md5" AS "banner_md5", "banner"."name" AS "banner_name", "banner"."type" AS "banner_type", "banner"."size" AS "banner_size", "banner"."comment" AS "banner_comment", "banner"."blurhash" AS "banner_blurhash", "banner"."properties" AS "banner_properties", "banner"."storedInternal" AS "banner_storedInternal", "banner"."url" AS "banner_url", "banner"."thumbnailUrl" AS "banner_thumbnailUrl", "banner"."webpublicUrl" AS "banner_webpublicUrl", "banner"."webpublicType" AS "banner_webpublicType", "banner"."accessKey" AS "banner_accessKey", "banner"."thumbnailAccessKey" AS "banner_thumbnailAccessKey", "banner"."webpublicAccessKey" AS "banner_webpublicAccessKey", "banner"."uri" AS "banner_uri", "banner"."src" AS "banner_src", "banner"."folderId" AS "banner_folderId", "banner"."isSensitive" AS "banner_isSensitive", "banner"."maybeSensitive" AS "banner_maybeSensitive", "banner"."maybePorn" AS "banner_maybePorn", "banner"."isLink" AS "banner_isLink", "banner"."requestHeaders" AS "banner_requestHeaders", "banner"."requestIp" AS "banner_requestIp", "reply"."id" AS "reply_id", "reply"."createdAt" AS "reply_createdAt", "reply"."replyId" AS "reply_replyId", "reply"."renoteId" AS "reply_renoteId", "reply"."threadId" AS "reply_threadId", "reply"."text" AS "reply_text", "reply"."name" AS "reply_name", "reply"."cw" AS "reply_cw", "reply"."userId" AS "reply_userId", "reply"."localOnly" AS "reply_localOnly", "reply"."renoteCount" AS "reply_renoteCount", "reply"."repliesCount" AS "reply_repliesCount", "reply"."reactions" AS "reply_reactions", "reply"."visibility" AS "reply_visibility", "reply"."uri" AS "reply_uri", "reply"."url" AS "reply_url", "reply"."fileIds" AS "reply_fileIds", "reply"."attachedFileTypes" AS "reply_attachedFileTypes", "reply"."visibleUserIds" AS "reply_visibleUserIds", "reply"."mentions" AS "reply_mentions", "reply"."mentionedRemoteUsers" AS "reply_mentionedRemoteUsers", "reply"."emojis" AS "reply_emojis", "reply"."tags" AS "reply_tags", "reply"."hasPoll" AS "reply_hasPoll", "reply"."channelId" AS "reply_channelId", "reply"."userHost" AS "reply_userHost", "reply"."replyUserId" AS "reply_replyUserId", "reply"."replyUserHost" AS "reply_replyUserHost", "reply"."renoteUserId" AS "reply_renoteUserId", "reply"."renoteUserHost" AS "reply_renoteUserHost", "renote"."id" AS "renote_id", "renote"."createdAt" AS "renote_createdAt", "renote"."replyId" AS "renote_replyId", "renote"."renoteId" AS "renote_renoteId", "renote"."threadId" AS "renote_threadId", "renote"."text" AS "renote_text", "renote"."name" AS "renote_name", "renote"."cw" AS "renote_cw", "renote"."userId" AS "renote_userId", "renote"."localOnly" AS "renote_localOnly", "renote"."renoteCount" AS "renote_renoteCount", "renote"."repliesCount" AS "renote_repliesCount", "renote"."reactions" AS "renote_reactions", "renote"."visibility" AS "renote_visibility", "renote"."uri" AS "renote_uri", "renote"."url" AS "renote_url", "renote"."fileIds" AS "renote_fileIds", "renote"."attachedFileTypes" AS "renote_attachedFileTypes", "renote"."visibleUserIds" AS "renote_visibleUserIds", "renote"."mentions" AS "renote_mentions", "renote"."mentionedRemoteUsers" AS "renote_mentionedRemoteUsers", "renote"."emojis" AS "renote_emojis", "renote"."tags" AS "renote_tags", "renote"."hasPoll" AS "renote_hasPoll", "renote"."channelId" AS "renote_channelId", "renote"."userHost" AS "renote_userHost", "renote"."replyUserId" AS "renote_replyUserId", "renote"."replyUserHost" AS "renote_replyUserHost", "renote"."renoteUserId" AS "renote_renoteUserId", "renote"."renoteUserHost" AS "renote_renoteUserHost", "replyUser"."id" AS "replyUser_id", "replyUser"."createdAt" AS "replyUser_createdAt", "replyUser"."updatedAt" AS "replyUser_updatedAt", "replyUser"."lastFetchedAt" AS "replyUser_lastFetchedAt", "replyUser"."lastActiveDate" AS "replyUser_lastActiveDate", "replyUser"."hideOnlineStatus" AS "replyUser_hideOnlineStatus", "replyUser"."username" AS "replyUser_username", "replyUser"."name" AS "replyUser_name", "replyUser"."followersCount" AS "replyUser_followersCount", "replyUser"."followingCount" AS "replyUser_followingCount", "replyUser"."notesCount" AS "replyUser_notesCount", "replyUser"."avatarId" AS "replyUser_avatarId", "replyUser"."bannerId" AS "replyUser_bannerId", "replyUser"."tags" AS "replyUser_tags", "replyUser"."isSuspended" AS "replyUser_isSuspended", "replyUser"."isSilenced" AS "replyUser_isSilenced", "replyUser"."isLocked" AS "replyUser_isLocked", "replyUser"."isBot" AS "replyUser_isBot", "replyUser"."isCat" AS "replyUser_isCat", "replyUser"."isAdmin" AS "replyUser_isAdmin", "replyUser"."isModerator" AS "replyUser_isModerator", "replyUser"."isExplorable" AS "replyUser_isExplorable", "replyUser"."isDeleted" AS "replyUser_isDeleted", "replyUser"."emojis" AS "replyUser_emojis", "replyUser"."host" AS "replyUser_host", "replyUser"."inbox" AS "replyUser_inbox", "replyUser"."sharedInbox" AS "replyUser_sharedInbox", "replyUser"."featured" AS "replyUser_featured", "replyUser"."uri" AS "replyUser_uri", "replyUser"."followersUri" AS "replyUser_followersUri", "replyUser"."showTimelineReplies" AS "replyUser_showTimelineReplies", "replyUser"."token" AS "replyUser_token", "replyUser"."driveCapacityOverrideMb" AS "replyUser_driveCapacityOverrideMb", "replyUserAvatar"."id" AS "replyUserAvatar_id", "replyUserAvatar"."createdAt" AS "replyUserAvatar_createdAt", "replyUserAvatar"."userId" AS "replyUserAvatar_userId", "replyUserAvatar"."userHost" AS "replyUserAvatar_userHost", "replyUserAvatar"."md5" AS "replyUserAvatar_md5", "replyUserAvatar"."name" AS "replyUserAvatar_name", "replyUserAvatar"."type" AS "replyUserAvatar_type", "replyUserAvatar"."size" AS "replyUserAvatar_size", "replyUserAvatar"."comment" AS "replyUserAvatar_comment", "replyUserAvatar"."blurhash" AS "replyUserAvatar_blurhash", "replyUserAvatar"."properties" AS "replyUserAvatar_properties", "replyUserAvatar"."storedInternal" AS "replyUserAvatar_storedInternal", "replyUserAvatar"."url" AS "replyUserAvatar_url", "replyUserAvatar"."thumbnailUrl" AS "replyUserAvatar_thumbnailUrl", "replyUserAvatar"."webpublicUrl" AS "replyUserAvatar_webpublicUrl", "replyUserAvatar"."webpublicType" AS "replyUserAvatar_webpublicType", "replyUserAvatar"."accessKey" AS "replyUserAvatar_accessKey", "replyUserAvatar"."thumbnailAccessKey" AS "replyUserAvatar_thumbnailAccessKey", "replyUserAvatar"."webpublicAccessKey" AS "replyUserAvatar_webpublicAccessKey", "replyUserAvatar"."uri" AS "replyUserAvatar_uri", "replyUserAvatar"."src" AS "replyUserAvatar_src", "replyUserAvatar"."folderId" AS "replyUserAvatar_folderId", "replyUserAvatar"."isSensitive" AS "replyUserAvatar_isSensitive", "replyUserAvatar"."maybeSensitive" AS "replyUserAvatar_maybeSensitive", "replyUserAvatar"."maybePorn" AS "replyUserAvatar_maybePorn", "replyUserAvatar"."isLink" AS "replyUserAvatar_isLink", "replyUserAvatar"."requestHeaders" AS "replyUserAvatar_requestHeaders", "replyUserAvatar"."requestIp" AS "replyUserAvatar_requestIp", "replyUserBanner"."id" AS "replyUserBanner_id", "replyUserBanner"."createdAt" AS "replyUserBanner_createdAt", "replyUserBanner"."userId" AS "replyUserBanner_userId", "replyUserBanner"."userHost" AS "replyUserBanner_userHost", "replyUserBanner"."md5" AS "replyUserBanner_md5", "replyUserBanner"."name" AS "replyUserBanner_name", "replyUserBanner"."type" AS "replyUserBanner_type", "replyUserBanner"."size" AS "replyUserBanner_size", "replyUserBanner"."comment" AS "replyUserBanner_comment", "replyUserBanner"."blurhash" AS "replyUserBanner_blurhash", "replyUserBanner"."properties" AS "replyUserBanner_properties", "replyUserBanner"."storedInternal" AS "replyUserBanner_storedInternal", "replyUserBanner"."url" AS "replyUserBanner_url", "replyUserBanner"."thumbnailUrl" AS "replyUserBanner_thumbnailUrl", "replyUserBanner"."webpublicUrl" AS "replyUserBanner_webpublicUrl", "replyUserBanner"."webpublicType" AS "replyUserBanner_webpublicType", "replyUserBanner"."accessKey" AS "replyUserBanner_accessKey", "replyUserBanner"."thumbnailAccessKey" AS "replyUserBanner_thumbnailAccessKey", "replyUserBanner"."webpublicAccessKey" AS "replyUserBanner_webpublicAccessKey", "replyUserBanner"."uri" AS "replyUserBanner_uri", "replyUserBanner"."src" AS "replyUserBanner_src", "replyUserBanner"."folderId" AS "replyUserBanner_folderId", "replyUserBanner"."isSensitive" AS "replyUserBanner_isSensitive", "replyUserBanner"."maybeSensitive" AS "replyUserBanner_maybeSensitive", "replyUserBanner"."maybePorn" AS "replyUserBanner_maybePorn", "replyUserBanner"."isLink" AS "replyUserBanner_isLink", "replyUserBanner"."requestHeaders" AS "replyUserBanner_requestHeaders", "replyUserBanner"."requestIp" AS "replyUserBanner_requestIp", "renoteUser"."id" AS "renoteUser_id", "renoteUser"."createdAt" AS "renoteUser_createdAt", "renoteUser"."updatedAt" AS "renoteUser_updatedAt", "renoteUser"."lastFetchedAt" AS "renoteUser_lastFetchedAt", "renoteUser"."lastActiveDate" AS "renoteUser_lastActiveDate", "renoteUser"."hideOnlineStatus" AS "renoteUser_hideOnlineStatus", "renoteUser"."username" AS "renoteUser_username", "renoteUser"."name" AS "renoteUser_name", "renoteUser"."followersCount" AS "renoteUser_followersCount", "renoteUser"."followingCount" AS "renoteUser_followingCount", "renoteUser"."notesCount" AS "renoteUser_notesCount", "renoteUser"."avatarId" AS "renoteUser_avatarId", "renoteUser"."bannerId" AS "renoteUser_bannerId", "renoteUser"."tags" AS "renoteUser_tags", "renoteUser"."isSuspended" AS "renoteUser_isSuspended", "renoteUser"."isSilenced" AS "renoteUser_isSilenced", "renoteUser"."isLocked" AS "renoteUser_isLocked", "renoteUser"."isBot" AS "renoteUser_isBot", "renoteUser"."isCat" AS "renoteUser_isCat", "renoteUser"."isAdmin" AS "renoteUser_isAdmin", "renoteUser"."isModerator" AS "renoteUser_isModerator", "renoteUser"."isExplorable" AS "renoteUser_isExplorable", "renoteUser"."isDeleted" AS "renoteUser_isDeleted", "renoteUser"."emojis" AS "renoteUser_emojis", "renoteUser"."host" AS "renoteUser_host", "renoteUser"."inbox" AS "renoteUser_inbox", "renoteUser"."sharedInbox" AS "renoteUser_sharedInbox", "renoteUser"."featured" AS "renoteUser_featured", "renoteUser"."uri" AS "renoteUser_uri", "renoteUser"."followersUri" AS "renoteUser_followersUri", "renoteUser"."showTimelineReplies" AS "renoteUser_showTimelineReplies", "renoteUser"."token" AS "renoteUser_token", "renoteUser"."driveCapacityOverrideMb" AS "renoteUser_driveCapacityOverrideMb", "renoteUserAvatar"."id" AS "renoteUserAvatar_id", "renoteUserAvatar"."createdAt" AS "renoteUserAvatar_createdAt", "renoteUserAvatar"."userId" AS "renoteUserAvatar_userId", "renoteUserAvatar"."userHost" AS "renoteUserAvatar_userHost", "renoteUserAvatar"."md5" AS "renoteUserAvatar_md5", "renoteUserAvatar"."name" AS "renoteUserAvatar_name", "renoteUserAvatar"."type" AS "renoteUserAvatar_type", "renoteUserAvatar"."size" AS "renoteUserAvatar_size", "renoteUserAvatar"."comment" AS "renoteUserAvatar_comment", "renoteUserAvatar"."blurhash" AS "renoteUserAvatar_blurhash", "renoteUserAvatar"."properties" AS "renoteUserAvatar_properties", "renoteUserAvatar"."storedInternal" AS "renoteUserAvatar_storedInternal", "renoteUserAvatar"."url" AS "renoteUserAvatar_url", "renoteUserAvatar"."thumbnailUrl" AS "renoteUserAvatar_thumbnailUrl", "renoteUserAvatar"."webpublicUrl" AS "renoteUserAvatar_webpublicUrl", "renoteUserAvatar"."webpublicType" AS "renoteUserAvatar_webpublicType", "renoteUserAvatar"."accessKey" AS "renoteUserAvatar_accessKey", "renoteUserAvatar"."thumbnailAccessKey" AS "renoteUserAvatar_thumbnailAccessKey", "renoteUserAvatar"."webpublicAccessKey" AS "renoteUserAvatar_webpublicAccessKey", "renoteUserAvatar"."uri" AS "renoteUserAvatar_uri", "renoteUserAvatar"."src" AS "renoteUserAvatar_src", "renoteUserAvatar"."folderId" AS "renoteUserAvatar_folderId", "renoteUserAvatar"."isSensitive" AS "renoteUserAvatar_isSensitive", "renoteUserAvatar"."maybeSensitive" AS "renoteUserAvatar_maybeSensitive", "renoteUserAvatar"."maybePorn" AS "renoteUserAvatar_maybePorn", "renoteUserAvatar"."isLink" AS "renoteUserAvatar_isLink", "renoteUserAvatar"."requestHeaders" AS "renoteUserAvatar_requestHeaders", "renoteUserAvatar"."requestIp" AS "renoteUserAvatar_requestIp", "renoteUserBanner"."id" AS "renoteUserBanner_id", "renoteUserBanner"."createdAt" AS "renoteUserBanner_createdAt", "renoteUserBanner"."userId" AS "renoteUserBanner_userId", "renoteUserBanner"."userHost" AS "renoteUserBanner_userHost", "renoteUserBanner"."md5" AS "renoteUserBanner_md5", "renoteUserBanner"."name" AS "renoteUserBanner_name", "renoteUserBanner"."type" AS "renoteUserBanner_type", "renoteUserBanner"."size" AS "renoteUserBanner_size", "renoteUserBanner"."comment" AS "renoteUserBanner_comment", "renoteUserBanner"."blurhash" AS "renoteUserBanner_blurhash", "renoteUserBanner"."properties" AS "renoteUserBanner_properties", "renoteUserBanner"."storedInternal" AS "renoteUserBanner_storedInternal", "renoteUserBanner"."url" AS "renoteUserBanner_url", "renoteUserBanner"."thumbnailUrl" AS "renoteUserBanner_thumbnailUrl", "renoteUserBanner"."webpublicUrl" AS "renoteUserBanner_webpublicUrl", "renoteUserBanner"."webpublicType" AS "renoteUserBanner_webpublicType", "renoteUserBanner"."accessKey" AS "renoteUserBanner_accessKey", "renoteUserBanner"."thumbnailAccessKey" AS "renoteUserBanner_thumbnailAccessKey", "renoteUserBanner"."webpublicAccessKey" AS "renoteUserBanner_webpublicAccessKey", "renoteUserBanner"."uri" AS "renoteUserBanner_uri", "renoteUserBanner"."src" AS "renoteUserBanner_src", "renoteUserBanner"."folderId" AS "renoteUserBanner_folderId", "renoteUserBanner"."isSensitive" AS "renoteUserBanner_isSensitive", "renoteUserBanner"."maybeSensitive" AS "renoteUserBanner_maybeSensitive", "renoteUserBanner"."maybePorn" AS "renoteUserBanner_maybePorn", "renoteUserBanner"."isLink" AS "renoteUserBanner_isLink", "renoteUserBanner"."requestHeaders" AS "renoteUserBanner_requestHeaders", "renoteUserBanner"."requestIp" AS "renoteUserBanner_requestIp" FROM "note" "note" INNER JOIN "user_list_joining" "userListJoining" ON "userListJoining"."userId" = "note"."userId"  INNER JOIN "user" "user" ON "user"."id"="note"."userId"  LEFT JOIN "drive_file" "avatar" ON "avatar"."id"="user"."avatarId"  LEFT JOIN "drive_file" "banner" ON "banner"."id"="user"."bannerId"  LEFT JOIN "note" "reply" ON "reply"."id"="note"."replyId"  LEFT JOIN "note" "renote" ON "renote"."id"="note"."renoteId"  LEFT JOIN "user" "replyUser" ON "replyUser"."id"="reply"."userId"  LEFT JOIN "drive_file" "replyUserAvatar" ON "replyUserAvatar"."id"="replyUser"."avatarId"  LEFT JOIN "drive_file" "replyUserBanner" ON "replyUserBanner"."id"="replyUser"."bannerId"  LEFT JOIN "user" "renoteUser" ON "renoteUser"."id"="renote"."userId"  LEFT JOIN "drive_file" "renoteUserAvatar" ON "renoteUserAvatar"."id"="renoteUser"."avatarId"  LEFT JOIN "drive_file" "renoteUserBanner" ON "renoteUserBanner"."id"="renoteUser"."bannerId" WHERE "userListJoining"."userListId" = $1 AND (("note"."visibility" = 'public' OR "note"."visibility" = 'home') OR "note"."userId" = $2 OR $3 = ANY("note"."visibleUserIds") OR $4 = ANY("note"."mentions") OR ("note"."visibility" = 'followers' AND ("note"."userId" IN (SELECT "following"."followeeId" AS "following_followeeId" FROM "following" "following" WHERE "following"."followerId" = $5) OR "note"."replyUserId" = $6)))) "distinctAlias" ORDER BY "distinctAlias"."note_id" DESC, "note_id" ASC LIMIT 11
misskey-web-1    | ERR  1	[api]	Internal error occurred in notes/user-list-timeline: canceling statement due to statement timeout
misskey-web-1    | WARN 1	[api]	SLOW API CALL DETECTED: notes/user-list-timeline (10033.614974975586ms)

## 📝 Steps to Reproduce

1. add timeline query as widget
2. create a rather big user list
3. select user list in the widget

## 📌 Environment

Misskey version: 12.119.0
Your OS: Ubuntu 20.04 (Docker)
Your browser: tested several browsers on OSX and Linux

**Comment by creamlike1024** [comment]
Similar issues with note/mentions.

Environment:
Debian 11.4 (docker)
Misskey 12.119.0

Web container logs:
```
misskey-web-1  | ERR  1	[api]	Internal error occurred in notes/mentions: canceling statement due to statement timeout
misskey-web-1  | WARN 1	[api]	SLOW API CALL DETECTED: notes/mentions (10008.984574005008ms)
```

db container logs:
```

misskey-db-1  | 2022-11-07 14:36:50.822 UTC [2497] ERROR:  canceling statement due to statement timeout
misskey-db-1  | 2022-11-07 14:36:50.822 UTC [2497] STATEMENT:  SELECT DISTINCT  ...(SQL statement, too long, ignored)

misskey-db-1  | 2022-11-07 14:36:50.826 UTC [2502] FATAL:  terminating connection due to administrator command
misskey-db-1  | 2022-11-07 14:36:50.826 UTC [2502] STATEMENT:  SELECT DISTINCT ...

misskey-db-1  | 2022-11-07 14:36:50.835 UTC [2501] FATAL:  terminating connection due to administrator command
misskey-db-1  | 2022-11-07 14:36:50.835 UTC [2501] STATEMENT:  SELECT DISTINCT ...


misskey-db-1  | 2022-11-07 14:36:50.863 UTC [1] LOG:  background worker "parallel worker" (PID 2502) exited with exit code 1
misskey-db-1  | 2022-11-07 14:36:50.864 UTC [1] LOG:  background worker "parallel worker" (PID 2501) exited with exit code 1
```

It seems misskey terminated SQL statement due to 10s timeout limit. Change `statement_timeout` in `misskey/packages/backend/src/postgre.ts` to a bigger value may work?

**Comment by creamlike1024** [comment]
Hello, I found the way to solve it.
This problem may be caused by the previous misskey process's failure to terminate the query normally, but the previous misskey PID is still locking the data.

I successfully solved the problem by taking the following steps:
1. Modified `postgre.ts`, set `statement_time` to a bigger value (such as 1000 * 60)
2. Rebuild images,restart container
3. Stop misskey web container (but do not stop postgresql container)
4. Enter postgresql shell (docker compose exec db bash)
5. Login into database with `psql -U misskey ` (replace your database username here)
6. Run SQL statement `SELECT * FROM pg_stat_activity WHERE state = 'active';` to find the pid
7. Kill the process by `SELECT pg_terminate_backend(<pid of the process>)`
8. Exit postgresql shell environment, start misskey web again

After these steps I successfully opened note/mentions, no error report in db log. But it still takes 24s to display the content, which results in a slow api call WARN. But I don't think that's a problem because my server performance is poor,and that's why it's still necessary to modified the `statement_timeout` value.
